src/lib/fixer.ts

```diff
@@ -131,6 +138,164 @@ export class _PmdFixGenerator extends FixGenerator {
             title: 'Clear Single Diagnostic',
             arguments: [this.document.uri, this.diagnostic]
         };
+
         return action;
     }
+
+    public generateClassLevelSuppression(): vscode.CodeAction {
+        // Find the end-of-line position of the class declaration where the diagnostic is found.
+        const classStartPosition = this.findClassStartPosition(this.diagnostic, this.document);
+        // const classEndOfLinePosition = 0;
+
+        const action = new vscode.CodeAction(messages.fixer.supressOnClass, vscode.CodeActionKind.QuickFix);
+        action.edit = new vscode.WorkspaceEdit();
+    
+        // Determine the appropriate suppression rule based on the type of diagnostic.code
+        let suppressionRule: string;
+        if (typeof this.diagnostic.code === 'object' && this.diagnostic.code !== null && 'value' in this.diagnostic.code) {
```

Comment:
Do you need both the `=== 'object'` and `!== null` check? Seems like the first would be sufficient. Also, in JS/TS it's typical to do `== null` instead of `=== null`, because `undefined == null` and usually you want to catch both null and undefined.

Reply:
Fixed.